### PR TITLE
fix(chore): fix centreon-ui typescript types warnings

### PR DESCRIPTION
**Pull request by karimsx**
The goal of this PR is to fix all warning caused by typescript typing system 

this pr must merged in same time with :  https://github.com/centreon/ui-context/pull/8

`npm run test` in centreon-ui

image

`npm run build` in any centreon module

image

Result :

`npm run test` in centreon-ui

image

`npm run build` in any centreon module

image

### CODE REVIEW

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — DISMISSED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -39,8 +38,8 @@ export { default as RegexpHelpTooltip } from './InputField/Search/RegexpHelpTool
 export { default as TextField } from './InputField/Text';
 
 export { default as SelectField,  } from './InputField/Select';
-import type {SelectEntry} from './InputField/Select';
-export type {SelectEntry}
+export type {SelectEntry} from './InputField/Select';
```
Comment (bdauria) [inline src/index.ts:41]: you need to run npm run eslint:fix

Comment (karimsx) [reply src/index.ts:41]: i have run eslint and eslint:fix but they show no errors?

Comment (Thebarda) [reply src/index.ts:41]: Did you run `npm run eslint:fix` on Windows ?
If yes, try to run the content of the script referenced in `package.json`

Comment (karimsx) [reply src/index.ts:41]: yes ofc, i did run `npm run eslint:fix` and `npm run eslint`, i will try aigain

Comment (Thebarda) [reply src/index.ts:41]: I mean this command: `eslint . --ext .js,.jsx,.ts,.tsx --fix`

Comment (karimsx) [reply src/index.ts]: fixed ;)

File: .eslintrc.js [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module.exports = {
   extends:
     './node_modules/@centreon/frontend-core/eslint/react/typescript.eslintrc.js',
+  root: true,
```
Comment (bdauria) [inline .eslintrc.js:4]: root: true is already included in the latest frontend-core

Comment (karimsx) [reply .eslintrc.js:4]: fixed in next commit

Comment (bdauria) [reply .eslintrc.js:4]: It seems still there :)

Comment (karimsx) [reply .eslintrc.js:4]: its the good one right now ;)

File: src/Listing/PaginationActions.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
+/* eslint-disable react/forbid-prop-types */
```
Comment (bdauria) [inline src/Listing/PaginationActions.js:1]: This file has been removed in previous commits

File: package.json [hunk 4]
```diff
@@ -112,6 +112,8 @@
     "ts-loader": "^8.0.18",
     "typescript": "^4.2.3",
     "webpack-merge": "^5.7.3",
-    "yup": "^0.32.9"
+    "yup": "^0.32.9",
+    "eslint-plugin-typescript-sort-keys": "1.x",
```
Comment (bdauria) [inline package.json:116]: use:
```
npm install eslint-plugin-typescript-sort-keys@latest --save-dev
npm install eslint-plugin-sort-keys-fix@latest --save-dev
```

instead

File: src/Listing/PaginationActions.js (hunk 3 above)
Comment (karimsx) [reply src/Listing/PaginationActions.js]: yes fixed right now

File: package.json (hunk 4 above)
Comment (karimsx) [reply package.json]: done ;)

**Review by bdauria — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**